### ISSUE TITLE
Device not seen in HA.

### ISSUE THREAD

**Issue by bert269** [KaufHA/common#4]
Followed the videos and instructions, but KAUF bulb is not seen in HA.
It was on firmware 1.851 and flashed 1.87.
Still did not see it in HA.

Then I tried the friendly name setup as presented in video instructions and flashed this code:
![image](https://github.com/KaufHA/common/assets/18601216/9467a56f-e4c1-49b6-83b0-02a48f1f4994)

no device is not accesable at all, not even by it's IP address.
How do I reset the bulb and access it now?

**Comment by bert269** [comment]
I managed to get the device reset by turning it of an on a few times, It came up with the default configureation and I was able to access it from 192.168.4.1. 
I renamed the device and also the yaml file. I re-flashed the firmware and this time it worked. It came up with the new name and I can access it from it's IP address on my wifi network.
![image](https://github.com/KaufHA/common/assets/18601216/66f244f4-fe06-4649-9129-06ec978dc71e)


I still do not see it in Home Assistant or in the ESPHOME integration.
What is the option: Kitchen Lamp No HASS OFF means? Does it mean HASS is on? 
I still am at a loss on how to add this to HA. The firware version is 1.879(y) and according to the website, 1.72 is visible in HA.
Second bulb that I am having issues with. 

Please advice. Thank you.

**Comment by bkaufx** [comment]
I don't see anything below.  Where are you at on this?  Once you get the bulb connected to the network, and able to access the bulb's web interface through typing its IP address in a web browser, then HA should find it.  If not, I would try adding it manually as described here: https://github.com/KaufHA/common#device-not-being-detected-by-home-assistant

**Comment by bert269** [comment]
Oooops sorry.
![image](https://github.com/KaufHA/common/assets/18601216/da56a817-07cd-451d-a9c4-5a9f5e134967)


Second device is back. I can access it via it's IP address, but still does not appear in HA.
Followed those instructions and also renamed the device.

Not in HA or even in ESPHome as a device.

First device is useless - does not even power back on after I tried the flash firmware 1.879 to it.

**Comment by bert269** [comment]
Yes, the bulb is: 192.168.1.92 and I can access it's web interface.
After the bulb has been configured it does not "appear in Home Assistant" as a device.
It does not show with the other ESPHOME devices as an entity on the integrations panel.
![image](https://github.com/KaufHA/common/assets/18601216/521582b4-637f-4150-80b8-018ca87714ef)

 - even after I set it up manually in ESPHOME using the following configuration:
![image](https://github.com/KaufHA/common/assets/18601216/deb42661-935c-4076-9ea1-2a751cf626f7)

On the bulb's web interface I see the controls and the log, with updates when I turn it on or off.

I am certain that this bulb is NOT in HA, it is not under the list of devices and not as a device in ESPHOME like my other four ESPHOME devices..

No - looking at the log on the bulb while rebooting Home Assistant - it does not mention anything about Home Asistant. It is not in home assistant. This is all I see after logging onto the device and rebooted Home Assistant:
![image](https://github.com/KaufHA/common/assets/18601216/7a0f9b38-dbca-49c6-8b52-f958e9e21eb2)

**Comment by bert269** [comment]
For somereason, the bulb finally showed up in HA. All that was different and changed, was the ESPHOME got updated today to the latest version. 
Trying to configure the bulb, created an 'invalid flow specified' error.
![image](https://github.com/KaufHA/common/assets/18601216/d0dead2a-be9f-4a98-9002-5cf63cb9712e)

**Comment by bkaufx** [comment]
The yaml filename really has nothing to do with anything.

I have never seen that invalid flow specified error from ESPHome and can't find any information about it.  

- Maybe try adding manually now that its seen by Home Assistant.
- try reflashing now that ESPHome has updated.
- Try reflashing without API encryption and change IP address too.

**Comment by bert269** [comment]
This is truly very, very strange that the bulb is acting like this. This
morning, I received a notification that 'a new device was found' - but when
i click on the 'more details' = nothing shows up.
It did this about three or four times - then it showed up.
I managed to configure it in HA - and YES, it works.


What the hell is going on - it looks like HA is sleeping or there is a huge
delay setting this up for this device.

OK - but let's move on - it is working and I hope it stays that way.
Thanks for your help.
<https://mssociety.donordrive.com/index.cfm?fuseaction=donorDrive.participant&participantID=52149&referrer=BF_emailbadge>

 Keep in TOUCH, with the American(-Dutch)
*Live free or die hard trying...*

*We may not be racing, but that doesn't mean I am not trying to beat you
;-)*
Bert
